libdnf/utils/locker.hpp

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_UTILS_LOCKER_HPP
+#define LIBDNF_UTILS_LOCKER_HPP
+
+#include <string>
+
+namespace libdnf::utils {
+
+constexpr int INVALID_FD = -1;
```

Comment:
Do we want `INVALID_FD` to be public? What to define it as a private member of the Locker class?

Reply:
Tbh. I don't see a reason to have this in a variable, just use a -1 directly?

Reply:
This is a special descriptor value expressing an invalid descriptor. Naming doesn't hurt anything. Of course definitely not a variable. But there is `constexpr` in the code. I have no problem with that.

Reply:
Sure... There's also an invalid descriptor expressed by a `-1` here: https://github.com/rpm-software-management/libdnf/blob/2f528454f867f409e28300f0c4a3d93f2fabac18/libdnf/utils/temp.hpp#L88

Shall we share the `INVALID_FD` definition to deduplicate? If not, I want to approach this consistently, so there should be an `INVALID_FD` constant in that file as well. Ultimately, `-1` is just simpler, it saves on LoCs, and there's no other difference.